AzureMonitorAgent/ama_tst/modules/install/check_ama.py

```diff
@@ -1,30 +1,41 @@
 import re
-import requests
 import xml.dom.minidom
+import urllib
+try:
+    import requests
+except ImportError:
+    pass
 
 from error_codes import *
 from errors      import error_info, get_input
 from helpers     import get_package_version
+from connect.check_endpts import check_internet_connect
 
 AMA_URL = 'https://docs.microsoft.com/en-us/azure/azure-monitor/agents/azure-monitor-agent-extension-versions'
 
 def get_latest_ama_version(curr_version):
+    # python2 and python3 compatible
     try:           
-        r = requests.get(AMA_URL)
-        tbody = r.text.split("<tbody>")[1].split("</tbody>")[0]
+        r = urllib.urlopen(AMA_URL).read()
+    except AttributeError:
+        r = requests.get(AMA_URL).text
+        
+    try:
+        tbody = r.split("<tbody>")[1].split("</tbody>")[0]
         tbody = "<tbody>" + tbody + "</tbody>"
-        with xml.dom.minidom.parseString(tbody) as dom:
```

Comment:
any reason why we stopped using the `with ... as ...`? generally this is good as it will automatically 'close' the `dom` object once the scope ends